drivers/gpu/drm/i915/gvt/mmio.c

```diff
@@ -213,6 +213,15 @@ int intel_vgpu_emulate_mmio_write(struct intel_vgpu *vgpu, uint64_t pa,
 	if (ret < 0)
 		goto err;
 
+	if (vgpu->entire_nonctxmmio_checked
+		&& intel_gvt_mmio_is_non_context(vgpu->gvt, offset)
+		&& vgpu_vreg(vgpu, offset)
+			!= *(u32 *)(vgpu->gvt->mmio.mmio_host_cache + offset)) {
```

Comment:
can we move gvt_host_reg to a common header file so that we can use it here?

Reply:
Sure. I planed to port my patch for 4.14 after it got  reviewed and merged.
I can combine them too. How do you think?

Reply:
I would like to combine them together to reduce the patch numbers, but it's up to you.